AWS CLI install instructions and credentials setup

gwsii opened the pull request:
Revised to break into three steps, adding link to script to install as well as link to instructions from AWS on how to install.

laurwang commented on Lesson0_BeforeWorkshop/README.md:
````diff
@@ -1,80 +1,44 @@
+From your workshop directory:
 ```sh
-<editor of choice> ~/.aws/credentials
+$ git clone https://github.com/Nordstrom/hello-retail-workshop.git
````
Should this be https://github.com/<your fork>/hello-retail-workshop.git?

erikerikson replied:
don't think so

laurwang replied:
oops.  I put in angle-bracket forked repo angle-bracket

erikerikson replied:
oh wait, if they've cloned it, then it would be to their cloned version and the following would make sense:
`https://github.com/<GitHubID>/hello-retail-workshop.git?`

laurwang replied:
to replace Nordstrom in the path, because we asked them to fork first

erikerikson replied:
😄

erikerikson replied:
and I was slow too

laurwang replied:
How come your angle bracket worked and mine didn't.  Do I need to escape it?

laurwang replied:
Ah, the strange quote mark, `<got it>`.

laurwang commented on Lesson0_BeforeWorkshop/README.md:
```diff
@@ -88,7 +52,7 @@ Regardless, install the serverless.com deployment framework - this will make it
 $ npm install -g serverless
```
Is 1.12 working?  Erik was worried about its breaking hello-retail, but haven't checked if all ok for hello-retail-workshop.

erikerikson replied:
Yes.  A patch was pushed 😄

laurwang commented on Lesson1_HelloRetailStream/README.md:
```diff
@@ -19,3 +19,27 @@ Look in the AWS console under Kinesis you should see your stream there as *Strea
 If you are a Nordstrom employee, paste the information into the #serverless-workshop channel.  We will add you to our fan-out lambda and you'll start seeing traffic as the workshop progresses.
+
+
+## Extra Credit: Using the AWS CLI to show Role and Stream ARNs
```
If this is Extra Credit, and we have an Extra Credit folder (Lesson 4) should we put this there, with a hyperlink to it?

gwsii replied:
Okay, yes I will link it.

gwsii replied:
Really just want these commands to be available to help us move quickly in the workshop if people come late, etc. Plus pushes the AWS CLI which I feel is a good way for people to fill in the "automation" gaps in Serverless.

gwsii commented:
>> Should this be here?
Well, I've put it here to capture the script primarily so that the instructor can either 1) send the student this script and they can run it, sending the output back (presumes student has AWS CLI installed) 2) instructor can log into their account, set their STAGE environment variable and run it.
>> re $STAGE
As stated, the teacher can set and then, if logged into the same account (which is possible, even likely for PCv2 SANDBOX accounts) then they will get the appropriate ARNs.

My objective is to provide then a script to the instructor that takes those ARNs as inputs.

Maybe I'm being confusing by keeping these changes coming in bit-by-bit.

Sorry about that. More to come, hope it starts making sense soon.

laurwang commented:
Sorry, will wait a bit and come back when it's in a complete state. 😄

gwsii commented:
Those are just backquotes (unshifted ~). What do you get when you use:
```sh
`echo Hello`
```

laurwang commented:
Is it my shell?  I get this
-bash: Hello: command not found

gwsii commented:
Well, technically those scripts are complete. They work and are potentially useful, just not perfect and no clear place for their introduction to whomever may need them.

gwsii commented:
I would expect it to work: https://unix.stackexchange.com/questions/27428/what-does-backquote-backtick-mean-in-commands

gwsii commented:
Does the script in the PR work for your (show-stream-and-role-arns.sh)?

gwsii commented:
>> Is it my shell? I get this
>> -bash: Hello: command not found

Actually, not that's right.

I meant to send you:
```
echo `echo Hello`
```

laurwang commented:
Yes!  After some trial and error, I figured that out, but I also figured out that the problem wasn't the backquote.  It was the fact that you have to get a unique result.
```
m00974110:~ bimx$ aws kinesis describe-stream --stream-name `aws kinesis list-streams | grep bimxOG | sed -n "s/^.*\"\(.*\)\".*/\1/p"` | grep StreamARN | sed -n "s/^.*\(arn\:aws\:kinesis\:.*Stream\).*/\1/p"
arn:aws:kinesis:us-west-2:468543742856:stream/bimxOGStream
m00974110:~ bimx$ aws kinesis describe-stream --stream-name `aws kinesis list-streams | grep bimx | sed -n "s/^.*\"\(.*\)\".*/\1/p"` | grep StreamARN | sed -n "s/^.*\(arn\:aws\:kinesis\:.*Stream\).*/\1/p"
usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

Unknown options: bimxStream
```
So in the case where you search for stage bimx, you'll also get bimxOG if it's lying around, because the search is too broad, and that blows it up.

gwsii commented:
Great BUG! 👍 Thanks for helping by running the script.

laurwang commented:
My pleasure, learned about scripting.  Learned to stop calling backquote "the strange quote", which was not winning me any style points.

gwsii commented:
We can continue to discuss where these script might live and how they can be invoked and support the activities of the instructor. Merging after addressing feedback so far.
